### PR TITLE
Added auto sync for birthdays and anniversaries on the app start (#906)

### PR DESCRIPTION
**Pull request by Aga-C**
Hi,

I've added an option for automatic sync of birthdays and anniversaries on the app start. It's optional, and there's a permission check while toggling an option. If user revokes the permission, it doesn't throw any error, just ignores adding new events.

https://user-images.githubusercontent.com/85929121/139481278-fd0bc1f7-28d3-44be-a9e7-ac1179be9f88.mp4

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt [hunk 1]
```diff
@@ -589,6 +593,37 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
         }
     }
 
+    private fun addBirthdaysAnniversaries() {
+        if (!hasPermission(PERMISSION_READ_CONTACTS)) {
+            return
+        }
+
+        val reminders = arrayListOf(REMINDER_OFF, REMINDER_OFF, REMINDER_OFF)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt:601]: the user can already select reminder at importing birthdays and anniversaries, lets reuse the values he selected here too

File: app/src/main/res/values-ar/strings.xml [hunk 2]
```diff
@@ -202,6 +202,7 @@
     <string name="highlight_weekends">ظلل نهايات الأسبوع في بعض العروض</string>
     <string name="allow_changing_time_zones">Allow changing event time zones</string>
     <string name="manage_quick_filter_event_types">Manage quick filter event types</string>
+    <string name="add_bdays_anniv_automatically">Add birthdays and anniversaries automatically after launching the app</string>
```
Comment (tibbi) [inline app/src/main/res/values-ar/strings.xml:205]: let's change this to "Add new birthdays automatically" and "Add new anniversaries automatically"

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt (hunk 1 above)
Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt:599]: People will expect having the last used reminders there. Just store the 3 reminders in some new shared pref config field.

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt [hunk 3]
```diff
@@ -621,6 +623,34 @@ class SettingsActivity : SimpleActivity() {
         }
     }
 
+    private fun setupAddBirthdaysAutomatically() {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt:626]: oh, this whole thing shouldnt be a settings item, just a checkbox in the importing dialog

Comment (Aga-C) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt:626]: Ok, so just to confirm:
- user taps on "Add contact birthdays" from menu
- dialog with setting reminders is displayed, and despite reminders user also has a checkbox "Add new birthdays automatically"
- if checkbox is selected, we also store the reminders setting in config
- rest works the same as now - user starts an app, and we check for new birthdays
Of course, the same logic would be applied to anniversaries too.
Am I right?

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt:626]: yep, like that

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt [hunk 4]
```diff
@@ -589,6 +591,41 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
         }
     }
 
+    private fun addBirthdaysAnniversariesAtStart() {
+        if (!hasPermission(PERMISSION_READ_CONTACTS)) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt:595]: you should check config.addAnniversariesAutomatically and config.addBirthdaysAutomatically first, those are a lot quicker and simpler checks than some permissions

**Comment by Aga-C** [comment]
I've applied changes.

image

**Comment by tibbi** [comment]
oh gosh, stop showing a toast message at every app launch, it is annoying

**Comment by Aga-C** [comment]
It shows toast only when there is a new anniversary or birthday in contacts. I've tested it again now, and it doesn't show every time.

**Comment by tibbi** [comment]
It was actually showing everytime to me with a success message. Guess it is some unrelated issue then, it doesnt recognize existing events or what. Will take a look at it again.

**Comment by tibbi** [comment]
okay I removed a corrupt contact, should be good I guess. Thanks

**Pull request merged**